core/src/main/scala/org/apache/spark/status/LiveEntity.scala

```diff
@@ -386,7 +386,7 @@ private class LiveExecutorStageSummary(
       metrics.memoryBytesSpilled,
       metrics.diskBytesSpilled,
       isExcluded,
-      Some(peakExecutorMetrics).filter(_.isSet),
+      Some(peakExecutorMetrics),
```

Comment:
why do we need this?

Reply:
Can we read `peakExecutorMetrics` as all 0 if it is none?

Reply:
> why do we need this?

Without this, data will not passed to restful API when there is no data, then in ui will show `NAN`

Reply:
I think it's better to resolve in the read side. Otherwise, there is no need to make this field optional as well

Reply:
> I think it's better to resolve in the read side. Otherwise, there is no need to make this field optional as well

The previous PR make this value as Option. 
Since currently, the way got executor summary data change
```
stageExecutorSummaryInfoKeys.forEach(function (columnKeyIndex) {
                    var executorSummary = responseBody.executorSummary[columnKeyIndex];
                    var executorDetail = executorDetailsMap[columnKeyIndex.toString()];
                    executorSummary.id = columnKeyIndex;
                    executorSummary.executorLogs = {};
                    executorSummary.hostPort = "CANNOT FIND ADDRESS";

                    if (executorDetail) {
                        if (executorDetail["executorLogs"]) {
                            responseBody.executorSummary[columnKeyIndex].executorLogs =
                                executorDetail["executorLogs"];
                            }
                        if (executorDetail["hostPort"]) {
                            responseBody.executorSummary[columnKeyIndex].hostPort =
                                executorDetail["hostPort"];
                        }
                    }
                    executorSummaryTable.push(responseBody.executorSummary[columnKeyIndex]);
                });
```

If not change this, throw exception since no `peakMemoryMetrics` value passed to restful api response 
![image](https://user-images.githubusercontent.com/46485123/101137533-f2dcc800-3649-11eb-8dc9-c74cadd7b616.png)

Reply:
If we want show this data in UI, pass a default value when no update happened on this metrics is also OK, right?

Reply:
IMO, make this value as Option seem to avoid convert null to json when there is no data

Reply:
Then we need to set the filed as non-optional in `StageData`, or change `ExecutorMetricsJsonDeserializer`

Reply:
> Then we need to set the filed as non-optional in `StageData`, or change `ExecutorMetricsJsonDeserializer`

If we change field as non-optional in `StageData`, we also need to change `ExecutorMetricsJsonDeserializer` too since it based on the field's type , and it was used in three places,
![image](https://user-images.githubusercontent.com/46485123/101142364-77cae000-3650-11eb-92b1-8e39531f36c6.png)
Just change here and remove filter may be the safest way.

Reply:
The code was introduce in https://github.com/apache/spark/pull/29020. 
I want to make sure the API is reasonable instead of being safest.

Reply:
See also: https://github.com/apache/spark/pull/30573/files#r535955674

Reply:
> See also: https://github.com/apache/spark/pull/30573/files#r535955674

Change to judge in stagepage.js